src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,132 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (!isIndexValid(index) && index != size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        if (size == elementData.length) {
+            elementData = grow();
+        }
```

Comment:
let's move this check to addElementInMiddle()(in add() method you are doing this anyway;

Reply:
done